src/Microsoft.Tye.Core/ProcessUtil.cs

```diff
@@ -109,7 +109,7 @@ public static class ProcessUtil
             process.BeginErrorReadLine();
 
             var cancelledTcs = new TaskCompletionSource<object?>();
-            using var _ = cancellationToken.Register(() => cancelledTcs.TrySetResult(null));
+            await using var _ = cancellationToken.Register(() => cancelledTcs.TrySetResult(null));
```

Comment:
Wow. I learned something new today thanks to this PR!

@davidfowl did you know about this? https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenRegistration.cs#L48

Reply:
do u want a PR that does CancellationTokens?

Reply:
Yah

Reply:
name for CancellationToken variables? my preference is `cancellation`, but you already have some with `cancellationToken`